src/utilities/dynamodb-helper.js

```diff
@@ -28,18 +49,44 @@ export const addChangesToUpdateParams = (params, changes, fieldsAllowedToUpdate)
   return params;
 };
 
-export const buildSoftDeleteUpdateParams = (item) => {
-  const eightWeeksAfter = moment().add({ weeks: 8, hour: 0 }); // hour: 0 for enforce correct precise time diff related to DST
+export const buildUpdateParamFromItem = (item, changes) => {
+  // NOTE: Use this method with caution as it bypass the field check
+  const baseParams = {
+    Key: {
+      InboundConversationId: item.InboundConversationId,
+      Layer: item.Layer
+    },
+    UpdateExpression: `SET UpdatedAt = :now`,
+    ExpressionAttributeValues: {
+      ':now': getUKTimestamp()
+    }
+  };
+  return addChangesToUpdateParams(baseParams, changes, Object.keys(changes));
+};
+
+export const buildParamsToClearPreviousOutboundRecord = item => {
+  let restoredStatus;
+  if (isConversation(item)) {
+    restoredStatus = ConversationStatus.INBOUND_COMPLETE;
+  } else if (isCore(item)) {
+    restoredStatus = CoreStatus.INBOUND_COMPLETE;
+  } else if (isFragment(item)) {
+    restoredStatus = FragmentStatus.INBOUND_COMPLETE;
+  } else {
+    throw new Error(`Got unexpected dynamodb item: ${item}`);
+  }
```

Comment:
Looks like a candidate for a switch? Though in the interest of time I'm happy to keep this as-is!

Reply:
Won't do now in interest of time.